src/Compiler/Microsoft.NET.Sdk.Razor.SourceGenerators/RazorSourceGenerator.cs

```diff
@@ -226,15 +226,28 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                 .Where(pair => pair.Right)
                 .Select((pair, _) => pair.Left);
 
+            var isAddComponentParameterAvailable = compilation.Select((compilation, _) =>
+            {
+                var renderTreeBuilder = compilation.GetTypeByMetadataName("Microsoft.AspNetCore.Components.Rendering.RenderTreeBuilder");
```

Comment:
`GetTypeByMetadataName` can fail for a number of reasons, but the most important one for us is that it can fail when a type is defined multiple times. We should instead use `GetTypesByMetadataName`, and then go through the results and ensure that one and only one is accessible to the current compilation. If there are none or more than one, we should suppress (the latter case may want a warning to be reported).

Reply:
Is there a scenario where we'd expect the type to come from the current assembly? If not, we should consider making this only recalculate when the assembly references change, like we do above for tag helpers.

Reply:
Agree on both points.